modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1890,15 +1886,15 @@ public boolean onUpdate(@NotNull WatchEvent evt) {
 
                     RaftGroupService partGrpSvc = internalTable.partitionRaftGroupService(partId);
 
-                    IgniteBiTuple<Peer, Long> leaderWithTerm = partGrpSvc.refreshAndGetLeaderWithTerm().join();
+                    LeaderWithTerm leaderWithTerm = partGrpSvc.refreshAndGetLeaderWithTerm().join();
 
                     // run update of raft configuration if this node is a leader
-                    if (localMember.name().equals(leaderWithTerm.get1().consistentId())) {
+                    if (localMember.name().equals(leaderWithTerm.leader().consistentId())) {
                         LOG.info("Current node={} is the leader of partition raft group={}. "
                                         + "Initiate rebalance process for partition={}, table={}",
                                 localMember.address(), replicaGrpId, partId, tbl.name());
 
-                        partGrpSvc.changePeersAsync(newNodes, leaderWithTerm.get2()).join();
+                        partGrpSvc.changePeersAsync(newNodes, List.of(), leaderWithTerm.term()).join();
```

Comment:
Is it ok that we always clear the list of learners here?

Reply:
that's ok for now, because nobody uses learners at the moment. I'm going to change this in my next PR

Reply:
Let's add a TODO to that next ticket, just in case